src/puppetlabs/puppetdb/queue.clj

```diff
@@ -30,15 +32,78 @@
     (catch Exception e
       (throw+ {:kind ::parse-error} e "Error parsing command"))))
 
+(defn sanitize-certname
+  "Replace any underscores and filename forbidden characters found in `certname`
+  with dashes."
+  [certname]
+  (let [forbidden-chars (conj constants/filename-forbidden-characters
+                              "_")]
+    (str/replace
+      certname
+      (re-pattern (str "("
+                       (->> forbidden-chars
+                           (map #(format "\\Q%s\\E" %)) ; escape regex chars
+                           (str/join "|"))
+                       ")"))
+      "-")))
+
+(defn max-certname-length
```

Comment:
Umm, let's pretend this never happened?  Agreed that it might not matter performance-wise, but regardless I was just wrongheaded here - thinking about a situation that doesn't match what we're *actually* doing (we don't keep the *utf-8* prefix bytes, they're regenerated elsewhere later), and so was handwaving about something that would just be more expensive than what you have.